cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -390,6 +390,25 @@ that have not been built" OFF)
          ON)
 endif()
 
+macro(validate_config)
+  foreach(category ${ARROW_OPTION_CATEGORIES})
+    set(option_names ${ARROW_${category}_OPTION_NAMES})
+
+    foreach(name ${option_names})
+      set(possible_values ${${name}_OPTION_POSSIBLE_VALUES})
+      set(value "${${name}}")
+      if(possible_values)
+        if(NOT ("${value}" IN_LIST possible_values))
```

Comment:
Could you skip this check if `NOT CMAKE_VERSION VERSION_LESS "3.3"` because `IN_LIST` doesn't exist in CMake 3.2?